src/ngx_http_lua_util.c

```diff
@@ -4266,4 +4266,25 @@ ngx_http_lua_set_sa_restart(ngx_log_t *log)
 #endif
 
 
+size_t
+ngx_http_lua_safe_header_filed_len(u_char *str, size_t len)
+{
+    u_char  *p;
+
+    /* search \r */
+    p = (u_char *) memchr(str, 13, len);
+    if (p != NULL) {
+        len = p - str;
+    }
+
+    /* search \n */
+    p = (u_char *) memchr(str, 10, len);
+    if (p != NULL) {
+        len = p - str;
+    }
```

Comment:
Since header values are usually quite short, maybe it is even faster to write a single tight loop to scan for `\r` and `\n` than 2 `memchr()` calls. We can do some micro benchmark if you have any doubts :)

Reply:
`memchr` is faster than `loop scan` in such case: https://gist.github.com/doujiang24/cbf071ad5b40d2bdf67b372732644676

1.2s vs 1.6s on my side.

Reply:
for the short strings, like the length shorter than 18, `loop` run faster, changed to use `loop` instead of `memchr`.